tests/nil_undefined_test.go

```diff
@@ -0,0 +1,72 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gopherjs/gopherjs/js"
+)
+
+func foo2(x interface{}) string {
+	s := fmt.Sprintln("x:", x)
+	s += fmt.Sprintln("x == nil:", x == nil)
+	s += fmt.Sprintln("x == js.Undefined:", x == js.Undefined)
+	return s
+}
+
+func TestNilUndefinedEquality(t *testing.T) {
+	if got, want := (interface{})(nil) == (interface{})(nil), true; got != want {
+		t.Errorf("\n(interface{})(nil) == (interface{})(nil): got %v, want %v", got, want)
+	}
+	if got, want := js.Undefined == js.Undefined, true; got != want {
+		t.Errorf("\njs.Undefined == js.Undefined: got %v, want %v", got, want)
+	}
+	var ui interface{} = js.Undefined
+	if got, want := ui == js.Undefined, true; got != want {
+		t.Errorf("\nui == js.Undefined: got %v, want %v", got, want)
+	}
+	var u = js.Undefined
+	if got, want := u == js.Undefined, true; got != want {
+		t.Errorf("\nu == js.Undefined: got %v, want %v", got, want)
+	}
+	if got, want := u == ui, true; got != want {
+		t.Errorf("\nu == ui: got %v, want %v", got, want)
+	}
+	if got, want := (interface{})(nil) == js.Undefined, false; got != want {
+		t.Errorf("\n(interface{})(nil) == js.Undefined: got %v, want %v", got, want)
+	}
+	if got, want := (interface{})(nil) == ui, false; got != want {
+		t.Errorf("\n(interface{})(nil) == ui: got %v, want %v", got, want)
+	}
+	if got, want := (interface{})(nil) == u, false; got != want {
+		t.Errorf("\n(interface{})(nil) == u: got %v, want %v", got, want)
+	}
```

Comment:
Some of these above may be already present everywhere, if so, they should be removed.